invenio_pidstore/cli.py

```diff
@@ -37,6 +40,28 @@
     from flask_cli import with_appcontext
 
 
+def loader_import(ctx, param, value):
+    """Import the loader."""
+    try:
+        return import_string(value)
+    except ImportError:
+        raise click.BadParameter('Loader {0} not found.'.format(value))
+
+
+def process_minter(ctx, param, value):
+    """Return the minter."""
+    with ctx.obj.load_app().app_context():
```

Comment:
What about returning a proxy object so we don't have to create the app context twice?

```
class LazyMinter(object):
    def __init__(self, value):
    # or even __init__(self, ctx, param, value):
        self.name = value
    @cached_property
    def minter(self):
        try:
            return current_pidstore.minters[value]
        except KeyError:
            raise click.BadParameter(
                param_hint='minter',
                message='Unknown minter {0}. Please use one of: {1}.'.format(
                    value, ', '.join(current_pidstore.minters.keys())
                )
            )

    def __call__(self, *args, **kwargs):
        return self.minter(*args, **kwargs)
```